### PR TITLE
[go-server] Feat: add required assertions to models

### PR DESCRIPTION
**Pull request by lwj5**
This PR adds required assertions for models.
- Checks a model if required fields are non-zero (tested)
- Checks a model's parent (recursive) if required fields are non-zero (tested)
- Checks a model's children (recursive) if required fields are non-zero (tested)
- Checks the model in an array (recursive e.g. slice in slice in slice) if required fields are non-zero (tested)
- Works with nullable and non-nullable types (tested)

This PR does not do assertion checks for:
- Map values (if they are a model)

Feel free to give suggestions, feedback!

Fixes:
A pointer of pointer?
```
	{{paramName}} := &{{dataType}}{}
	if err := json.NewDecoder(r.Body).Decode(&{{paramName}}); err != nil {
```
I removed the pointer from the first line.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.3.x`, `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by lwj5** [comment]
Hi @antihax @grokify @kemokemo @jirikuncar @ph4r5h4d, please help to take a look at this PR

**Comment by lwj5** [comment]
Tests failure unrelated 
`Plugin org.openapitools:openapi-generator-maven-plugin:5.2.0-SNAPSHOT or one of its dependencies could not be resolved`

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/GoServerCodegen.java [hunk 1]
```diff
@@ -106,6 +106,12 @@ public GoServerCodegen() {
         optAddResponseHeaders.defaultValue(addResponseHeaders.toString());
         cliOptions.add(optAddResponseHeaders);
 
+        // option to disallow unknown fields in body
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/GoServerCodegen.java:109]: Suggestion: elaborating a bit more by providing an example on what this option does.

Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/GoServerCodegen.java:109]: And this option seems to be related to `disallowAdditionalPropertiesIfNotPresent`.

Can we reuse `disallowAdditionalPropertiesIfNotPresent` instead?

https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenConstants.java#L385-L388

Comment (lwj5) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/GoServerCodegen.java]: Thanks for the feedback. Reverted, and fixed the API.

**Comment by lwj5** [comment]
On-hold. I think I might be able to add assertions for Nested arrays (array-in-array).

**Review by wing328 — APPROVED** [review]
Tested locally and didn't spot any issue.

**Pull request merged**